app/src/main/java/com/spascoding/taskycourse/feature_auth/presentation/util/UserDataValidator.kt

```diff
@@ -0,0 +1,56 @@
+package com.spascoding.taskycourse.feature_auth.presentation.util
```

Comment:
Validating such user data is rather business logic, so presentation might not be the right place for it. If you work with a domain layer, that's the right place. Otherwise, data.